manim/mobject/coordinate_systems.py

```diff
@@ -710,6 +717,482 @@ def prepare_for_nonlinear_transform(self, num_inserted_curves=50):
         return self
 
 
+class PolarPlane(Axes):
+    r"""Creates a polar plane with background lines.
+
+    Parameters
+    ----------
+    azimuth_step
+        The number of divisions in the azimuth (also known as the `angular coordinate` or `polar angle`). If ``None`` is specified then it will use the default
+        specified by ``azimuth_units``:
+
+        - ``"PI radians"`` or ``"TAU radians"``: 20
+        - ``"degrees"``: 36
+        - ``"gradians"``: 40
+        - ``None``: 1
+
+        A non-integer value will result in a partial division at the end of the circle.
+
+    size
+        The diameter of the plane.
+
+    radius_step
+        The distance between faded radius lines.
+
+    radius_max
+        The maximum value of the radius.
+
+    azimuth_units
+        Specifies a default labelling system for the azimuth. Choices are:
+
+        - ``"PI radians"``: Fractional labels in the interval :math:`\left[0, 2\pi\right]` with :math:`\pi` as a constant.
+        - ``"TAU radians"``: Fractional labels in the interval :math:`\left[0, \tau\right]` (where :math:`\tau = 2\pi`) with :math:`\tau` as a constant.
+        - ``"degrees"``: Decimal labels in the interval :math:`\left[0, 360\right]` with a degree (:math:`^{\circ}`) symbol.
+        - ``"gradians"``: Decimal lables in the interval :math:`\left[0, 400\right]` with a superscript "g" (:math:`^{g}`).
+        - ``None``: Decimal labels in the interval :math:`\left[0, 1\right]`.
+
+    azimuth_compact_fraction
+        If the ``azimuth_units`` choice has fractional labels, choose whether to combine the constant in a compact form :math:`\tfrac{xu}{y}` as opposed to :math:`\tfrac{x}{y}u`, where :math:`u` is the constant.
+
+    azimuth_offset
+        The angle offset of the azimuth, expressed in radians.
+
+    azimuth_direction
+        The direction of the azimuth.
+
+        - ``"CW"``: Clockwise.
+        - ``"CCW"``: Anti-clockwise.
+
+    azimuth_label_buff
+        The buffer for the azimuth labels.
+
+    azimuth_label_scale
+        The scale of the azimuth labels.
+
+    radius_config
+        The axis config for the radius.
+
+    Examples
+    --------
+
+    .. manim:: PolarPlaneExample
+        :ref_classes: PolarPlane
+        :save_last_frame:
+
+        polarplane_pi = PolarPlane(azimuth_units="PI radians", size=6, azimuth_label_scale=0.75, radius_config={'number_scale_value': 0.75}).add_coordinates()
+
+        class PolarPlaneExample(Scene):
+            def construct(self):
+                self.add(polarplane_pi)
```

Comment:
To keep consistent with the rest of the docs, polarplane_pi should be defined within the `class PolarPlaneExample`. Also, `polarplane_pi` requires a lot of horizontal scrolling to see fully in the docs, so running black on this example outside of the docstring would be nice